x-pack/plugins/security_solution/server/endpoint/routes/trusted_apps/utils.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ExceptionListItemSchema } from '../../../../../lists/common/shared_exports';
+import { TrustedApp } from '../../../../common/endpoint/types';
+
+/**
+ * Map an ExcptionListItem to a TrustedApp item
+ * @param exceptionListItem
+ */
+export const exceptionItemToTrustedAppItem = (
+  exceptionListItem: ExceptionListItemSchema
+): TrustedApp => {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const { entries, description, created_by, created_at, name, _tags, id } = exceptionListItem;
+  const os = osFromTagsList(_tags);
+  return {
+    entries,
+    description,
+    created_at,
+    created_by,
+    name,
+    os,
+    id,
+  };
+};
+
+/**
+ * Retrieves the OS entry from a list of tags (property returned with ExcptionListItem).
+ * For Trusted Apps each entry must have at MOST 1 OS.
+ * */
+const osFromTagsList = (tags: string[]): string => {
+  for (const tag of tags) {
+    if (tag.startsWith('os:')) {
+      return tag;
+    }
+  }
+  return '';
```

Comment:
I know it's not "supposed" to happen, but I'd rather fail here than hide the error, there is no meaningful handling of this in UI.

This is always a problem of using generic structures to represent more specific concepts - they leak through and break invariance. Not trying to trigger same discussion again :D just it would be so nicer to just use SavedObject ;)

Actually again a bit on the topic of our own API interface instead of inheriting ExceptionListItem - we now scatter transformations: OS is extracted on the server while path/hash/signer will have to be extracted on the client. I'd again suggest to not send entries but rather translate it to the data structure we can directly consume in UI. This is also why I mentioned in grooming that I would not incorporate signer in API for now.

Reply:
could we have it return `unknown` at least so the UI could use it if needed?

Reply:
Ok. I will set it to a string `unknown`, but that will likely cause issues once we introduce a type for it like an enum.

Reply:
`'unknown'` feels like a magic string waiting to happen (until it is an explicit enum case). Magic strings should not happen from the server. If the front-end ever wanted to display missing OS's differently (like a blank column), one would have to filter for the string value, which is a bit code-smelly. I didn't look up the call chain where this is used, but can this return `string | null`? The missing case, being `null`. That would perhaps offer the front-end an opportunity to display/handle explicitly. At the cost of null-checking, which also isn't fun, but more common.

Stop me if this is anti-typescript behavior.